vignettes/formats_precedence.Rmd

```diff
@@ -0,0 +1,300 @@
+
+```{r}
+lyt2 <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  analyze(vars = "AGE", afun = function(x) {
+    in_rows(
+      "Mean" = rcell(mean(x), format = "xx.xxx"),
+      .formats = "xx.xx"
+    )
+  })
+
+build_table(lyt2, ADSL)
+```
+
+### Parent table format and inheritance.
+
+In addition to the cell level, the format can be specified at the parent table level(s). Then, if no format has been set by the user for a cell, the format from the nearest parent split table will be inherited (if any).
+
+```{r}
+lyt3 <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  analyze(vars = "AGE", mean, format = "xx.x")
+
+build_table(lyt3, ADSL)
+```
+
+If the cell format was specified too, then the parent table format is ignored for this cell, since the cell format is treated as the most specific.
+
+```{r}
+lyt4 <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  analyze(
+    vars = "AGE", afun = function(x) {
+      rcell(mean(x), format = "xx.xx", label = "Mean")
+    },
+    format = "xx.x"
+  )
+
+build_table(lyt4, ADSL)
+
+lyt4a <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  analyze(
+    vars = "AGE", afun = function(x) {
+      in_rows(
+        "Mean" = rcell(mean(x)),
+        "SD" = rcell(sd(x)),
+        .formats = "xx.xx"
+      )
+    },
+    format = "xx.x"
+  )
+
+build_table(lyt4a, ADSL)
+```
+
+In the following, slightly more complicated example, we can observe partial inheritance. That is, only `SD` cell inherits the parent table's format, while the `Mean` cell does not.
+
+```{r}
+lyt5 <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  analyze(
+    vars = "AGE", afun = function(x) {
+      in_rows(
+        "Mean" = rcell(mean(x), format = "xx.xx"),
+        "SD" = rcell(sd(x))
+      )
+    },
+    format = "xx.x"
+  )
+
+build_table(lyt5, ADSL)
+```
+
+## `NA` Handling
+
+Consider the following layout and the resulting table created.
+
+```{r}
+lyt6 <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  split_rows_by("SEX") %>%
+  analyze(vars = "AGE", afun = mean, format = "xx.xx")
+
+build_table(lyt6, ADSL)
+```
+
+In the output, the cell corresponding to `UNDIFFERENTIATED` level of `SEX` and `B: Placebo drug` level of `ARM` categorical variables is displayed as `NA`. This happened because the cell's value was `NA`. `rtables` allows user to specify a string which should be displayed when formatted if the cell's value(s) are all `NA`. Similar to the configuration of formatting for numbers, user can specify the `NA` string in a few different places, at the **cell** level or **parent table** level. The `NA` string precedence and inheritance rules are the same as those for number formats precedence, described in the previous section of this vignette. We will illustrate them with a few examples.
```

Comment:
```suggestion
In the output, the cell corresponding to `UNDIFFERENTIATED` level of `SEX` and `B: Placebo drug` level of `ARM` categorical variables is displayed as `NA`. This happened because there were no values for the mean to calculate that respected these two conditions. `rtables` allows the user to specify a string to display when cell values are `NA`. Similar to formats for numbers, the user can specify a string to replace `NA` with the parameter `format_na_str` or `.format_na_str`. This can be specified at the **cell** or **parent table** level. The `NA` string precedence and inheritance rules are the same as those for number formats precedence, described in the previous section of this vignette. We will illustrate them with a few examples.
```